consensus/src/lib.rs

```diff
@@ -294,12 +246,12 @@ impl Consensus {
         }
 
         // If the certificate was in the previous fallback wave and there was a commit, then it is
-        // in the fallback
-        let fb_leader = self.try_fallback_commit(&certificate, max(fb_wave - 1, 1), state);
+        // in the current steady state wave
+        let fb_leader = self.try_fallback_commit(&certificate, fb_wave - 1, state);
```

Comment:
Can't this underflow?

Reply:
No, since if fb_wave = 0 then either ss_wave = 0 or ss_wave = 1. If ss_wave is 0 then the function returns None from if statement from before. If ss_wave = 1 then the mode is already determined since all validators begin from steady state wave 1, and so the function will return None. I will add an explicit underflow check to be safe.